src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -0,0 +1,127 @@
+﻿using CommunityToolkit.Maui.Views;
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui;
+
+/// <inheritdoc cref="IPopupService"/>
+public class PopupService : IPopupService
+{
+	readonly IServiceProvider serviceProvider;
+
+	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
+
+	/// <summary>
+	/// Creates a new instance of <see cref="PopupService"/>.
+	/// </summary>
+	/// <param name="serviceProvider">The <see cref="IServiceProvider"/> implementation.</param>
+	public PopupService(IServiceProvider serviceProvider)
+	{
+		this.serviceProvider = serviceProvider;
+	}
```

Comment:
I recommend using the existing references to `IServiceProvider`. This removes the requirement for the dev to inject `IServiceProvider`:

```suggestion
	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();

	static IServiceProvider serviceProvider =>
#if WINDOWS
			MauiWinUIApplication.Current.Services;
#elif ANDROID
            MauiApplication.Current.Services;
#elif IOS || MACCATALYST
			MauiUIApplicationDelegate.Current.Services;
#else
			throw InvalidOperationException($"{nameof(IServiceProvider)} requires a platform specific implementation");
#endif
```

Reply:
Is this better/cleaner? I really can't decide... `IServiceProvider` is registered automatically for us by .NET MAUI but it might be nice to remove the constructor...